pinot-tools/src/main/resources/log4j2.xml

```diff
@@ -39,6 +39,25 @@
       </Policies>
       <DefaultRolloverStrategy max="10"/>
     </RollingFile>
+    <Routing name="taskLogs" ignoreExceptions="false">
+      <Routes pattern="${ctx:taskId}">
+        <Route>
+          <RollingFile
+              name="minionTaskLogs-${ctx:taskId}"
+              fileName="${env:LOG_ROOT}/minionTaskLogs-${ctx:taskId:-default}.log"
+              filePattern="${env:LOG_ROOT}/minionTaskLogs-%d{yyyy-MM-dd}-%i.log"
+              immediateFlush="true">
+            <PatternLayout pattern="${env:LOG_PATTERN}"/>
+            <Policies>
+              <SizeBasedTriggeringPolicy size="19500KB"/>
+            </Policies>
+            <DefaultRolloverStrategy max="50"/>
+          </RollingFile>
+        </Route>
+      </Routes>
+      <!-- Created appender TTL -->
+      <IdlePurgePolicy timeToLive="15" timeUnit="minutes"/>
```

Comment:
i understood the rollover if > 19.5MB or rollover if > 50 files. 
What's the 15 minutes TTL? will that be enough? 
Also is 50 max good enough? We could have a lot more tasks running at once rt?

Reply:
per the doc: 15min TTL is used to close the dynamically created appender, after it's idle for up to TTL. 

as to purge old log files, there are multiple options: by max count as in the example here or time based (which may be more suitable for prod use, e.g. to keep log files as long as the task info is kept in helix, which is 24hr by default)